tests/integration/targets/mso_backup/tasks/main.yml

```diff
@@ -49,464 +64,814 @@
+      register: query_non_existent_tenant
+
+    - name: Verify non existent Tenant
+      assert:
+        that:
+          - query_non_existent_tenant is not changed
+          - query_non_existent_tenant.current == {}
+      when: version.current.version is version('2.2.4e', '!=')
+
+    # Query a Backup
+    - name: Query ansibleBackup3
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup3
+        state: query
+      register: query_ansibleBackup3
+
+    - name: Verify query_ansibleBackup3
+      assert:
+        that:
+          - query_ansibleBackup3 is not changed
+          - query_ansibleBackup3.current | length == 2
+
+    # Query All
+    - name: Query All
+      mso_backup:
+        <<: *mso_info
+        state: query
+      register: query_all_backups
+
+    - name: Verify query_all_backups
+      assert:
+        that:
+          - query_all_backups is not changed
+          - query_all_backups.current | selectattr("name", "match", "^ansibleBackup.*") | list | length == 7
+
+    # Remove Backup
+    - name: Remove ansibleBackup1 in check mode
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup1
+        state: absent
+      check_mode: yes
+      register: cm_remove_ansibleBackup1
+
+    - name: Verify cm_remove_ansibleBackup1
+      assert:
+        that:
+          - cm_remove_ansibleBackup1 is changed
+
+    - name: Remove ansibleBackup1 in normal mode
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup1
+        state: absent
+      register: nm_remove_ansibleBackup1
+
+    - name: Verify nm_remove_ansibleBackup1
+      assert:
+        that:
+          - nm_remove_ansibleBackup1 is changed
+
+    - name: Remove ansibleBackup1 in normal mode again
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup1
+        state: absent
+      register: nm_remove_ansibleBackup1_2
+
+    - name: Verify nm_remove_ansibleBackup1_2
+      assert:
+        that:
+          - nm_remove_ansibleBackup1_2 is not changed
+
+    - name: Remove ansibleBackup3 in normal mode
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup3
+        state: absent
+      ignore_errors: yes
+      register: nm_remove_ansibleBackup3
+
+    - name: Verify nm_remove_ansibleBackup3
+      assert:
+        that:
+          - nm_remove_ansibleBackup3.msg is match ("Multiple backups with same name found. Existing backups with similar names{{':'}} ansibleBackup3_[0-9]*, ansibleBackup3_[0-9]*")
+
+    - name: Query non_existent_backup
+      mso_backup:
+        <<: *mso_info
+        backup: nonExistentBackup
+        state: query
+      register: query_non_existent_backup
+
+    - name: Verify query_non_existent_backup
+      assert:
+        that:
+          - query_non_existent_backup is not changed
+
+# Backup validation tasks for MSO/NDO version > 3.2
+- name: Execute tasks only for MSO version > 3.2
+  when: version.current.version is version('3.2', '>')
+  block:
```

Comment:
Why do we need a complete separate one for v3.2+ most of the existing task above could apply for both right? (if you improved them we could run your new tasks for both too)